session/handler_test.go

```diff
@@ -255,3 +257,54 @@ func TestSessionLogout(t *testing.T) {
 		}
 	})
 }
+
+func TestSessionRequest(t *testing.T) {
+	conf, reg := internal.NewFastRegistryWithMocks(t)
+
+	// Start kratos server
+	publicTS, adminTS := testhelpers.NewKratosServerWithCSRF(t, reg)
+
+	mockServerURL := urlx.ParseOrPanic(publicTS.URL)
+
+	conf.MustSet(config.ViperKeyAdminBaseURL, adminTS.URL)
+	testhelpers.SetDefaultIdentitySchema(t, conf, "file://./stub/identity.schema.json")
+	testhelpers.SetIdentitySchemas(t, conf, map[string]string{
+		"customer": "file://./stub/handler/customer.schema.json",
+		"employee": "file://./stub/handler/employee.schema.json",
+	})
+	conf.MustSet(config.ViperKeyPublicBaseURL, mockServerURL.String())
+
+	session := func(t *testing.T, base *httptest.Server, href string, expectCode int) AdminIdentitySessionResponse {
+		req, err := http.NewRequest("GET", base.URL+href, nil)
+		require.NoError(t, err)
+
+		res, err := base.Client().Do(req)
+		require.NoError(t, err)
+
+		require.EqualValues(t, expectCode, res.StatusCode)
+		defer res.Body.Close()
+
+		var apiRes AdminIdentitySessionResponse
+		err = json.NewDecoder(res.Body).Decode(&apiRes)
+		require.NoError(t, err)
+		require.Contains(t, res.Header.Get("set-cookie"), config.DefaultSessionCookieName)
+
+		return apiRes
+	}
+
+	t.Run("case=should return 200 after successful session creation", func(t *testing.T) {
```

Comment:
Similarly to the test case above that says `should return 202 after invalidating all sessions`, that's not all what happens in `case=should return 200 after successful session creation`.

In the existing test above, the test also checks if there aren't any sessions in the DB. The test that you're adding is also checking whether the session is returned and if it is valid and active. I'd suggest updating our test case to something like
```suggestion
	t.Run("case=should return 200 after successful session creation", func(t *testing.T) {
```


```suggestion
	t.Run("case=when creating a session should respond with 200 and an active and valid token", func(t *testing.T) {
```

Reply:
Perhaps you've added the description before implementing the test? One technique I've been using is calling the test case something like "TODO" before implementing it fully.

This forces me to rename it only after I've implemented it, which results in a name better matching what actually happens.

This is a bit of a nitpick here since the test actually checks for MORE than it says it does, but it can become an issue if a test case does LESS than it says it does - here's when people can get into a false sense of security when making changes and reviewing the existing tests.

Reply:
Changed name to `case=should return 200 after successful session creation and return valid session and token`